src/Microsoft.DotNet.Interactive.Notebooks/Notebook/NotebookFileFormatHandler.cs

```diff
@@ -26,30 +25,35 @@ static NotebookFileFormatHandler()
             _serializerOptions.Converters.Add(new DataDictionaryConverter());
         }
 
-        public static NotebookDocument Parse(string fileName, byte[] rawData, string defaultLanguage, IDictionary<string, string> kernelLanguageAliases)
+        public static NotebookDocument Parse(string fileName, string notebookContent, string defaultLanguage,
+            IDictionary<string, string> kernelLanguageAliases)
         {
             var extension = Path.GetExtension(fileName);
             switch (extension.ToLowerInvariant())
             {
                 case ".dib":
                 case ".dotnet-interactive":
-                    return ParseInteractiveNotebook(rawData, defaultLanguage, kernelLanguageAliases);
+                    return ParseInteractiveNotebook(notebookContent, defaultLanguage, kernelLanguageAliases);
                 case ".ipynb":
-                    return ParseJupyterNotebook(rawData, kernelLanguageAliases);
+                    return ParseJupyterNotebook(notebookContent, kernelLanguageAliases);
                 default:
                     throw new NotSupportedException($"Unable to parse a notebook document of type '{extension}'");
             }
         }
 
-        private static NotebookDocument ParseInteractiveNotebook(byte[] rawData, string defaultLanguage, IDictionary<string, string> kernelLanguageAliases)
+        public static NotebookDocument Parse(string fileName, byte[] rawData, string defaultLanguage, IDictionary<string, string> kernelLanguageAliases)
         {
-            // `markdown` and its alias `md` aren't cell languages that the kernel is aware of, but nevertheless something that needs to be parsed like it is
-            var allLanguageAliases = ImmutableDictionary<string, string>.Empty
-                .AddRange(kernelLanguageAliases)
-                .Add("markdown", "markdown")
-                .Add("md", "markdown");
-
             var content = Encoding.UTF8.GetString(rawData);
+            return Parse(fileName, content, defaultLanguage, kernelLanguageAliases);
+        }
+
+        public static NotebookDocument ParseInteractiveNotebook(string content, string defaultLanguage, IDictionary<string, string> kernelLanguageAliases)
```

Comment:
There's some inconsistent naming here with `Parse` versus `ParseInteractiveNotebook`.

The model might be a little inside-out. Something more explicit and open-ended could be like:

```cs
NotebookDocument a = DibParser.Parse(...); // overloads supporting files or content
NotebookDocument a = JupyterNotebookParser.Parse(...); // overloads supporting files or content
NotebookDocument b = TryDotNetMarkdownParser.Parse(...); // overloads supporting files or content

```

The last one makes me wonder if `NotebookDocument` is the right name.